### PR TITLE
cardano-sl-networking test suite fixes and improvements

### PR DESCRIPTION
**Pull request by avieth**
## Description

1. It ~~not~~ now compiles
2. It's much faster (there was an incorrect `awaitSTM` function)
3. The `Peers` quickcheck generator respects that types invariants (formerly it did not)
4. Removed some unused modules

## Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by erikd** [comment]
Have a test suite and not running it in CI is a bad idea. Lets get this fixed as a high priority.

**Comment by avieth** [comment]
> Have a test suite and not running it in CI is a bad idea. Lets get this fixed as a high priority.

Ideally we'd run it only if there were actually a change to `cardano-sl-networking`, which is comparatively rare.

**Comment by erikd** [comment]
> Ideally we'd run it only if there were actually a change to `cardano-sl-networking`

Or anything that it depends on, like `cardano-sl-util`. It is possible for changes to `util` to break something in `networking` :)

File: networking/test/Test/Util.hs [hunk 1]
```diff
@@ -76,31 +73,29 @@ import           Node (Conversation (..), ConversationActions (..), Listener (..
                        nodeId, simpleNodeEndPoint)
 import           Node.Conversation (Converse)
 import           Node.Message.Binary (BinaryP, binaryPacking)
+import           Pos.Util.Trace (wlogTrace)
 
 -- | Run a computation, but kill it if it takes more than a given number of
 --   Microseconds to complete. If that happens, log using a given string
 --   prefix.
+--
+--   TODO use System.Timeout?
```
Comment (erikd) [inline networking/test/Test/Util.hs]: Yes, probably a good idea.

**Comment by erikd** [comment]
```
networking/test/Test/Util.hs:42:1: warning: [-Wunused-imports]
    The import of ‘concurrently’
    from module ‘Control.Concurrent.Async’ is redundant

networking/test/Test/Util.hs:45:1: warning: [-Wunused-imports]
    The import of ‘newTVarIO, writeTVar’
    from module ‘Control.Concurrent.STM.TVar’ is redundant

networking/test/Test/Util.hs:46:1: warning: [-Wunused-imports]
    The import of ‘MonadCatch’
    from module ‘Control.Exception.Safe’ is redundant

networking/test/Test/Util.hs:59:1: warning: [-Wunused-imports]
    The import of ‘Mockable.Class’ is redundant
      except perhaps to import instances from ‘Mockable.Class’
    To import instances alone, use: import Mockable.Class()

networking/test/Test/NodeSpec.hs:49:9: warning: [-Wunused-local-binds]
    Defined but not used: ‘tcpTransportUnbounded’

```
and more.

**Comment by avieth** [comment]
> Actually, even with this patch I'm still getting build failures in the networking tests when I run stack test on the whole project.

Ok will try that myself.

Ah yes, there are different `-W` settings when you build the whole project versus building a particular part.

**Comment by avieth** [comment]
@erikd I think I've fixed the redundant things.

**Comment by avieth** [comment]
> I'm approving the PR conditional on the warnings being fixed :)

OK will do. @erikd do you approve as well? I used to be able to merge with at least 1 approval. New policy?

**Comment by parsonsmatt** [comment]
I'm seeing more warnings:

```haskell
Preprocessing test suite 'cardano-sl-networking-test' for
cardano-sl-networking-1.1.1...
[2 of 5] Compiling Test.NodeSpec    ( test/Test/NodeSpec.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/cardano-sl-networking-test/cardano-sl-networking-test-tmp/Test/NodeSpec.o )

/home/matt/Projects/cardano-sl/networking/test/Test/NodeSpec.hs:58:16: warning: [-Wtype-defaults]
    • Defaulting the following constraint to type ‘[Char]’
        Data.String.IsString t0
          arising from the literal ‘"TCP unbounded queueing"’
    • In the expression: "TCP unbounded queueing"
      In the first argument of ‘seq’, namely
        ‘("TCP unbounded queueing", tcpTransportUnbounded)’
      In the expression:
        ("TCP unbounded queueing", tcpTransportUnbounded)
        `seq`
          ("TCP one-place queueing", tcpTransportOnePlace)
          `seq`
            ("TCP fair queueing", tcpTransportFair)
            `seq` ("In-memory", memoryTransport)

/home/matt/Projects/cardano-sl/networking/test/Test/NodeSpec.hs:59:16: warning: [-Wtype-defaults]
    • Defaulting the following constraint to type ‘[Char]’
        Data.String.IsString t0
          arising from the literal ‘"TCP one-place queueing"’
    • In the expression: "TCP one-place queueing"
      In the first argument of ‘seq’, namely
        ‘("TCP one-place queueing", tcpTransportOnePlace)’
      In the second argument of ‘seq’, namely
        ‘("TCP one-place queueing", tcpTransportOnePlace)
         `seq`
           ("TCP fair queueing", tcpTransportFair)
           `seq` ("In-memory", memoryTransport)’

/home/matt/Projects/cardano-sl/networking/test/Test/NodeSpec.hs:60:16: warning: [-Wtype-defaults]
    • Defaulting the following constraint to type ‘[Char]’
        Data.String.IsString t0
          arising from the literal ‘"TCP fair queueing"’
    • In the expression: "TCP fair queueing"
      In the first argument of ‘seq’, namely
        ‘("TCP fair queueing", tcpTransportFair)’
      In the second argument of ‘seq’, namely
        ‘("TCP fair queueing", tcpTransportFair)
         `seq` ("In-memory", memoryTransport)’
[3 of 5] Compiling Test.Network.Broadcast.OutboundQueueSpec ( test/Test/Network/Broadcast/OutboundQueueSpec.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/cardano-sl-networking-test/cardano-sl-networking-test-tmp/Test/Network/Broadcast/OutboundQueueSpec.o )
Linking .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/cardano-sl-networking-test/cardano-sl-networking-test ...
Installing library in
/home/matt/Projects/cardano-sl/.stack-work/install/x86_64-linux/lts-9.1/8.0.2/lib/x86_64-linux-ghc-8.0.2/cardano-sl-networking-1.1.1-L2kLkASK6ldH4Z3BtLVS0o
Installing executable(s) in
/home/matt/Projects/cardano-sl/.stack-work/install/x86_64-linux/lts-9.1/8.0.2/bin
Registering cardano-sl-networking-1.1.1...

--  End of log file: /home/matt/Projects/cardano-sl/.stack-work/logs/cardano-sl-networking-1.1.1.log
```

**Comment by avieth** [comment]
> I'm seeing more warnings:

Added a type sig. Should be fixed now.

**Comment by parsonsmatt** [comment]
I was running into more compiler errors/warnings when I was building this. `develop` has broken tests now that are failing my `stack test` builds, so I'm going to merge this in when CI is green :smile:

**Pull request merged**